### PR TITLE
Mar kolya/handle scopes

### PR DESCRIPTION
**Pull request by mar-kolya**

### CODE REVIEW

File: dd-java-agent/instrumentation/aws-java-sdk-2.2/src/main/java8/datadog/trace/instrumentation/aws/v2/TracingExecutionInterceptor.java [hunk 1]
```diff
@@ -33,9 +33,11 @@
   @Override
   public void beforeExecution(
       final Context.BeforeExecution context, final ExecutionAttributes executionAttributes) {
-    final Span span = GlobalTracer.get().buildSpan("aws.command").start();
-    DECORATE.afterStart(span);
-    executionAttributes.putAttribute(SPAN_ATTRIBUTE, span);
+    try (final Scope scope = GlobalTracer.get().buildSpan("aws.command").startActive(false)) {
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/aws-java-sdk-2.2/src/main/java8/datadog/trace/instrumentation/aws/v2/TracingExecutionInterceptor.java]: Please create the span first and use activate instead.

File: dd-java-agent/instrumentation/couchbase-2.0/src/main/java/datadog/trace/instrumentation/couchbase/client/CouchbaseBucketInstrumentation.java [hunk 2]
```diff
@@ -113,14 +114,16 @@ public void call() {
...
-                  .buildSpan("couchbase.call")
-                  .withTag(DDTags.RESOURCE_NAME, resourceName)
-                  .withTag("bucket", bucket)
-                  .start()));
+      try (final Scope scope =
+          GlobalTracer.get()
+              .buildSpan("couchbase.call")
+              .withTag(DDTags.RESOURCE_NAME, resourceName)
+              .withTag("bucket", bucket)
+              .startActive(false)) {
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/couchbase-2.0/src/main/java/datadog/trace/instrumentation/couchbase/client/CouchbaseBucketInstrumentation.java]: Please create the span first and call activate.

**Comment by mar-kolya** [comment]
@tylerbenson This has been updated, could you please have another look?

File: dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/ChannelFutureListenerInstrumentation.java [hunk 3]
```diff
@@ -83,18 +83,19 @@ public static TraceScope activateScope(@Advice.Argument(0) final ChannelFuture f
...
           GlobalTracer.get()
               .buildSpan("netty.connect")
               .withTag(Tags.COMPONENT.getKey(), "netty")
               .start();
-      Tags.ERROR.set(errorSpan, true);
-      errorSpan.log(singletonMap(ERROR_OBJECT, cause));
-      errorSpan.finish();
+      try (final Scope scope = GlobalTracer.get().scopeManager().activate(errorSpan, true)) {
+        Tags.ERROR.set(errorSpan, true);
+        errorSpan.log(singletonMap(ERROR_OBJECT, cause));
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/ChannelFutureListenerInstrumentation.java]: Looks like I missed a decorator...  Please change to use `onError` and `beforeFinish`

File: dd-java-agent/instrumentation/netty-4.1/src/main/java/datadog/trace/instrumentation/netty41/ChannelFutureListenerInstrumentation.java [hunk 4]
```diff
@@ -83,18 +83,19 @@ public static TraceScope activateScope(@Advice.Argument(0) final ChannelFuture f
...
 
       final Span errorSpan =
           GlobalTracer.get()
               .buildSpan("netty.connect")
               .withTag(Tags.COMPONENT.getKey(), "netty")
               .start();
-      Tags.ERROR.set(errorSpan, true);
-      errorSpan.log(singletonMap(ERROR_OBJECT, cause));
-      errorSpan.finish();
+      try (final Scope scope = GlobalTracer.get().scopeManager().activate(errorSpan, true)) {
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/netty-4.1/src/main/java/datadog/trace/instrumentation/netty41/ChannelFutureListenerInstrumentation.java]: I'm not going to make you change it now, but going forward please just assume that the boolean argument goes away (ie, set it to false) and that you must manually close the span.

File: dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/ChannelFutureListenerInstrumentation.java (hunk 3 above)
Comment (mar-kolya) [reply dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/ChannelFutureListenerInstrumentation.java]: Should be fixed now

File: dd-java-agent/instrumentation/netty-4.1/src/main/java/datadog/trace/instrumentation/netty41/ChannelFutureListenerInstrumentation.java (hunk 4 above)
Comment (mar-kolya) [reply dd-java-agent/instrumentation/netty-4.1/src/main/java/datadog/trace/instrumentation/netty41/ChannelFutureListenerInstrumentation.java]: I went ahead and fixed places I could find.

File: dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/ChannelFutureListenerInstrumentation.java [hunk 5]
```diff
@@ -90,9 +90,10 @@ public static TraceScope activateScope(@Advice.Argument(0) final ChannelFuture f
               .buildSpan("netty.connect")
               .withTag(Tags.COMPONENT.getKey(), "netty")
               .start();
-      try (final Scope scope = GlobalTracer.get().scopeManager().activate(errorSpan, true)) {
-        Tags.ERROR.set(errorSpan, true);
-        errorSpan.log(singletonMap(ERROR_OBJECT, cause));
+      try (final Scope scope = GlobalTracer.get().scopeManager().activate(errorSpan, false)) {
+        NettyHttpServerDecorator.DECORATE.onError(errorSpan, cause);
+        NettyHttpServerDecorator.DECORATE.beforeFinish(errorSpan);
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/ChannelFutureListenerInstrumentation.java]: Please static import the `DECORATE`.

Comment (mar-kolya) [reply dd-java-agent/instrumentation/netty-4.0/src/main/java/datadog/trace/instrumentation/netty40/ChannelFutureListenerInstrumentation.java]: It seems to be useful to explicitly show that we are using server decorator.

**Pull request merged**